contracts/multisig-prover/src/encoding/rkyv/mod.rs

```diff
@@ -11,14 +12,16 @@ use std::{array::TryFromSliceError, collections::BTreeMap};
 
 use crate::{error::ContractError, payload::Payload};
 
+use super::add_27;
+
 type Result<T> = core::result::Result<T, ContractError>;
 
-pub fn encode(
+pub fn encode_execute_data(
     signers_with_sigs: Vec<SignerWithSig>,
     payload_hash: [u8; 32],
     verifier_set: &VerifierSet,
     payload: &Payload,
-) -> Result<HexBinary> {
+) -> error_stack::Result<HexBinary, ContractError> {
```

Comment:
Aligning APi's with repo usage, making our rkyv module to match the signatures from other encoding modules for the sake of predictability.